partiql-eval/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerExcludeTests.kt

```diff
@@ -0,0 +1,1069 @@
+                <<
+                    {
+                        'a': 1, -- to be excluded
+                        'a': 2, -- to be excluded
+                        'a': 3, -- to be excluded
+                        'b': 4,
+                        'c': 5
+                    }
+                >> AS t
+                """,
+                """<<{'b': 4, 'c': 5}>>"""
+            ),
+            EvaluatorTestCase( // EXCLUDE with non-existent paths; no error
+                """
+                SELECT *
+                EXCLUDE t.path_does_not_exist, t.path_does_not_exist.*.foo, t.a['does not exist'], t.a
+                FROM 
+                <<{'a': 1, 'b': 2}>> AS t   -- only exclude `a`
+                """,
+                """<<{'b': 2}>>"""
+            ),
+            EvaluatorTestCase( // EXCLUDE with different FROM source bindings
+                """
+                SELECT *
+                EXCLUDE t.a[*].bar, t.a.bar, t.a.*.bar  -- EXCLUDE all `bar`
+                FROM 
+                <<
+                    {'a': [{'foo': 0, 'bar': 1, 'baz': 2}, {'foo': 3, 'bar': 4, 'baz': 5}]},
+                    {'a': {'foo': 6, 'bar': 7, 'baz': 8}},
+                    {'a': {'a1': {'foo': 9, 'bar': 10, 'baz': 11}, 'a2': {'foo': 12, 'bar': 13, 'baz': 14}}}
+                >> AS t
+                """,
+                """
+                <<
+                    {'a': [{'foo': 0, 'baz': 2}, {'foo': 3, 'baz': 5}]}, 
+                    {'a': {'foo': 6, 'baz': 8}}, 
+                    {'a': {'a1': {'foo': 9, 'baz': 11}, 'a2': {'foo': 12, 'baz': 14}}}
+                >>
+                """
+            ),
+            EvaluatorTestCase( // EXCLUDE with GROUP BY, HAVING, and aggregation
+                """
+                SELECT SUM(t.b) AS total, g
+                EXCLUDE g[*].t.c                    -- EXCLUDE
+                FROM
+                [
+                    {'a': 1, 'b': 2, 'c': 3},       -- group `t.a` = 1
+                    {'a': 1, 'b': 22, 'c': 33},     -- group `t.a` = 1
+                    {'a': 2, 'b': 222, 'c': 333}    -- row/group omitted due to `HAVING` clause
+                ] AS t
+                GROUP BY t.a GROUP AS g
+                HAVING COUNT(t.a) > 1
+                """,
+                """
+                <<
+                    {
+                        'total': 24,    -- `total` from row 1's `a` (2) + row 2's `a` (22) = 24
+                        'g': <<{'t': {'a': 1, 'b': 2}}, {'t': {'a': 1, 'b': 22}}>>  -- `EXCLUDE`s `c`
+                    }
+                >>
+                """
+            ),
+        )
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(ExcludeTests::class)
+    fun validExcludeTests(tc: EvaluatorTestCase) = runEvaluatorTestCase(
+        tc = tc.copy(targetPipeline = EvaluatorTestTarget.PARTIQL_PIPELINE),
+        session = EvaluationSession.standard()
+    )
+
+// TODO ALAN: add back subsumption tests (following rebase and refactor)
```

Comment:
After #1249 is merged in, the compiled exclude expressions will be refactored to allow for better reuse between `EvaluatingCompiler` and the `PhysicalPlanCompiler`. Through that refactor, these tests will be added back.